File: Python/Program/Mahasiswa.py
from SivitasAkademik import SivitasAkademik


class Mahasiswa(SivitasAkademik):

    __nim = ""
    __prodi = ""
    __fakultas = ""

    def __init__(self, nik="", nama="", gender="", asal_universitas="", email_edu="", nim="", prodi = "", fakultas=""):
        super().__init__(nik, nama, gender, asal_universitas, email_edu)
        self.__nim = nim
        self.__fakultas = fakultas
        self.__prodi = prodi

    def set_nim(self, nim):
        self.__nim = nim

    def get_nim(self):
        return self.__nim

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

    def get_fakultas(self):
        return self.__fakultas

    def display_data(self):
        super().display_data()
        print("| NIM                 : ", self.__nim)
        print("| Fakultas            : ", self.__fakultas)
        print("| [Agr] Program Studi : ", self.__prodi.kode_prodi, " - ", self.__prodi.nama_prodi)

File: CPP/Program/Mahasiswa.cpp
using namespace std;
#include <bits/stdc++.h>
#include "SivitasAkademik.cpp"
class Mahasiswa : public SivitasAkademik
{
private:
    string nim;
    string prodi;
    string fakultas;

public:
    // Constructor
    Mahasiswa()
    {
        this->nim = "";
        this->prodi = "";
        this->fakultas = "";
    }

    Mahasiswa(string nik, string nama, string gender, string asal_universitas, string email_edu, string nim, string prodi, string fakultas) : SivitasAkademik(nik, nama, gender, asal_universitas, email_edu)
    {
        this->nim = nim;
        this->prodi = prodi;
        this->fakultas = fakultas;
    }

    // Setter & Getter

    void setNim(string nim)
    {
        this->nim = nim;
    }

    string getNim()
    {
        return nim;
    }

    void setProdi(string prodi)
    {
        this->prodi = prodi;
    }

    string getProdi()
    {
        return prodi;
    }

    void setFakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }

    string getFakultas()
    {
        return fakultas;
    }

    void displayDaya()
    {
        SivitasAkademik::displayDaya();
        cout << "| NIM              : " << this->getNim() << endl;
        cout << "| Program Studi    : " << this->getProdi() << endl;
        cout << "| Fakultas         : " << this->getFakultas() << endl;
    }

    // Destructor
    ~Mahasiswa() {}
};